src/lib/common/github/issuereporter.ts

```diff
@@ -0,0 +1,223 @@
+        }
+
+        return result;
+    }
+
+    private async openIssue(issueData: IssueData) {
+        const labels: Array<string> = [
+            this.getScanLabel(issueData.scan),
+            this.getErrorTypeLabel(issueData.errorType)
+        ];
+
+        /* istanbul ignore else */
+        if (production) {
+            labels.push('production');
+        }
+
+        /* istanbul ignore if */
+        if (environment === 'browser') {
+            labels.push('browser');
+        }
+
+        const env = environment === undefined ? ' ' : ` [${environment}] `;
+
+        await this.octokit.issues.create(Object.assign(
+            {},
+            this.GITHUB_DATA,
+            {
+                body: this.getErrorMessage(issueData),
+                labels,
+                title: `[${this.getEmoji(issueData.errorType)}]${env}${issueData.url}`
+            }
+        ));
+    }
+
+    private async searchIssues(q: string) {
+        const result = await this.octokit.search.issues({ q });
+
+        return result.data.items;
+    }
+
+    public async report(issueData: IssueData) {
+
+        // Get open issues for a given URL.
+        /*
+         * Note: Search returns 100 results per page, but
+         *       the query shouldn't return so many results.
+         */
+        const issues = await this.searchIssues(`${issueData.url} in:title is:open repo:${this.GITHUB_DATA.owner}/${this.GITHUB_DATA.repo}`);
+
+        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        /*
+         * If there are no problems in the latest scan run with
+         * the given URL, close any existing issues related to it.
+         */
+
+        if (!issueData.errorType) {
+            for (const issue of issues) {
+                await this.closeIssue(issue);
+            }
+
+            return;
+        }
+
+        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        // If there were problems with the URL:
+
+        /*
+         * 1) If there is already an issue opened for the same problem,
+         *    add a new comment to that issue and update the labels.
+         */
+
+        for (const issue of issues) {
+
+            const issueLabels = issue.labels.map((label) => {
+                return label.name;
+            }) || [];
+
+            if (issueLabels.includes(this.getErrorTypeLabel(issueData.errorType))) {
+                await this.addIssueComment(issue, issueData);
+                await this.updateIssueLabels(issue, issueLabels.concat(this.getScanLabel(issueData.scan)));
+
+                return;
+            }
+        }
+
+        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        // 2) Otherwise open a new issue.
+
+        await this.openIssue(issueData);
+    }
+
+    private async updateIssueLabels(issue: IssuesUpdateParams, labels: string[]) {
+        await this.editIssue({
+            labels,
+            number: issue.issue_number
```

Comment:
`number` is deprecated. Use `issue_number` instead.
```suggestion
            issue_number: issue.issue_number
```